airflow/jobs/scheduler_job_runner.py

```diff
@@ -1847,3 +1851,23 @@ def _executor_to_tis(self, tis: list[TaskInstance]) -> dict[BaseExecutor, list[T
             _executor_to_tis[ExecutorLoader.load_executor(executor)].append(ti)
 
         return _executor_to_tis
+
+    @provide_session
+    def _check_and_activate_dagrun_scheduling(self, session: Session = NEW_SESSION):
+        tis = session.scalars(select(TI).where(TI.state == TaskInstanceState.UP_FOR_RETRY)).all()
```

Comment:
This pattern worries me. What if we have thousands of UP_FOR_RETRY's, we're going to try and check them all every 5 seconds?

Reply:
I felt it would be difficult to have that many tasks in the up_for_retry state simultaneously. Won't it?

Reply:
I don't think so?

```
one_day_of_seconds = 60 * 60 * 24
with DAG("retrier", schedule=None):
    for i in range(0, 1 * 1000):
        BashOperator(
            task_id=f"task_{i}",
            retry_delay=one_day_of_seconds,
            retries=1,
            bash_command="exit 1",
        )
```

Reply:
I mean, in a prod environment, it would be difficult to have many TIs in retry state